### PR TITLE
Bump version; make boost conditional

### PR DESCRIPTION
**Pull request by raovgarimella**
Bump up Jali version to 1.1.6
Make Boost dependency conditional only for older versions

### CODE REVIEW

**Comment by scheibelp** [comment]
Close/reopen to restart CI

**Comment by raovgarimella** [comment]
Is the problem in my code - I couldn't quite see the issue.

**Comment by raovgarimella** [comment]
@scheibelp Hi Peter, Should I do something to fix my package so that the tests pass or is this an issue with the testing setup? Thanks

File: var/spack/repos/builtin/packages/jali/package.py [hunk 1]
```diff
@@ -31,7 +32,8 @@ class Jali(CMakePackage):
 
     depends_on('mpi')
 
-    depends_on('boost')
+    if version < Version('1.1.6'):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/jali/package.py]: (EDIT: see next comment, I misread this) @alalazo pointed out that (EDIT: I misunderstood what he was suggesting and the below is the correct way to do it) ~this needs to be `if self.spec.version < Version('1.1.6')`, or - more idiomatic to spack - `if self.spec.satisfies('@:1.1.5')` (I prefer the latter)~

Comment (scheibelp) [reply var/spack/repos/builtin/packages/jali/package.py]: Wait scratch all that this occurs outside of the method: so it needs to be

```
depends_on('boost', when='@:1.1.5')
```

(and remove the `if`)

Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: ok. Thanks

**Review by scheibelp — APPROVED** [review]

**Pull request merged**